src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Simple.cs

```diff
@@ -0,0 +1,573 @@
+
+                    PipeTransmissionMode readMode = server.ReadMode;
+                    Assert.Throws<InvalidOperationException>(() => server.Read(buffer, 0, buffer.Length));
+                    Assert.Throws<InvalidOperationException>(() => server.ReadByte());
+                    Assert.Throws<InvalidOperationException>(() => { server.ReadAsync(buffer, 0, buffer.Length); });
+                }
+
+                Assert.Throws<InvalidOperationException>(() => server.Disconnect());    // disconnect when not connected 
+                Assert.Throws<InvalidOperationException>(() => server.IsMessageComplete);
+            }
+        }
+
+        [Fact]
+        public void OperationsOnUnconnectedClient()
+        {
+            using (var pair = CreateNamedPipePair())
+            {
+                var client = pair.clientStream;
+                byte[] buffer = new byte[] { 0, 0, 0, 0 };
+
+                if (client.CanRead)
+                {
+                    Assert.Throws<InvalidOperationException>(() => client.Read(buffer, 0, buffer.Length));
+                    Assert.Throws<InvalidOperationException>(() => client.ReadByte());
+                    Assert.Throws<InvalidOperationException>(() => { client.ReadAsync(buffer, 0, buffer.Length); });
+                    Assert.Throws<InvalidOperationException>(() => client.ReadMode);
+                    Assert.Throws<InvalidOperationException>(() => client.ReadMode = PipeTransmissionMode.Byte);
+                }
+                if (client.CanWrite)
+                {
+                    Assert.Throws<InvalidOperationException>(() => client.Write(buffer, 0, buffer.Length));
+                    Assert.Throws<InvalidOperationException>(() => client.WriteByte(5));
+                    Assert.Throws<InvalidOperationException>(() => { client.WriteAsync(buffer, 0, buffer.Length); });
+                }
+
+                Assert.Throws<InvalidOperationException>(() => client.NumberOfServerInstances);
+                Assert.Throws<InvalidOperationException>(() => client.TransmissionMode);
+                Assert.Throws<InvalidOperationException>(() => client.InBufferSize);
+                Assert.Throws<InvalidOperationException>(() => client.OutBufferSize);
+                Assert.Throws<InvalidOperationException>(() => client.SafePipeHandle);
+            }
+        }
+
+        [Fact]
+        public void DisposedServerPipe_Throws_ObjectDisposedException()
+        {
+            using (var pair = CreateNamedPipePair())
+            {
+                var pipe = pair.serverStream;
+                pipe.Dispose();
+                byte[] buffer = new byte[] { 0, 0, 0, 0 };
+
+                Assert.Throws<ObjectDisposedException>(() => pipe.Disconnect());
+                Assert.Throws<ObjectDisposedException>(() => pipe.WaitForConnection());
+                Assert.ThrowsAsync<ObjectDisposedException>(async () => { await pipe.WaitForConnectionAsync(); });
```

Comment:
This should be

    await Assert.ThrowsAsync<ObjectDisposedException>(() => pipe.WaitForConnectionAsync());

`ThrowsAsync` doesn't unwrap Task<Task> but return Task which could be awaited.